src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,245 @@
+                           OperationEnvironment& env)
+{
+  if (index == 0) // Invalid
+    return false;
+
+  // Request
+  EosRequestFlightInfo data;
+  data.flight_id = ToLE16(static_cast<uint16_t>(index));
+  data.crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&data), sizeof(data) - 1, 0xFF, 0x69);
+  std::span<std::byte> message(reinterpret_cast<std::byte*>(&data),
+                               sizeof(data));
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  // Read response (ACK byte already read)
+  EosFlightInfoResponse info;
+  try {
+    port.FullRead(
+      { reinterpret_cast<std::byte*>(&info) + 1, sizeof(info) - 1 },
+      env,
+      std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  if (Calculate8bitCRC(
+        reinterpret_cast<uint8_t*>(&info), sizeof(info), 0xFF, 0x69) !=
+      0x00) // Check CRC
+    return false;
+
+  /* Flight ID for downloading is the index (1 = newest),
+  not the ID from this message */
+  flight.internal.lx_eos.flight_id = index;
+  flight.date = BrokenDate::FromJulianDate(FromLE32(info.uiDate));
+  flight.start_time = BrokenTime::FromSecondOfDay(FromLE32(info.uiTakeOff));
+  flight.end_time = BrokenTime::FromSecondOfDay(FromLE32(info.uiLanding));
+  flight.internal.lx_eos.file_size = FromLE32(info.size);
+  return true;
+}
+
+AllocatedArray<std::byte>
+LXEosDevice::GetFlightLogBlock(uint16_t flight_id,
+                               uint16_t block_id,
+                               OperationEnvironment& env)
+{
+  AllocatedArray<std::byte> block;
+
+  // Request
+  EosRequestFlightBlock data;
+  data.flight_id = ToLE16(flight_id);
+  data.block_id = ToLE16(block_id);
+  data.crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&data), sizeof(data) - 1, 0xFF, 0x69);
+  std::span<std::byte> message(reinterpret_cast<std::byte*>(&data),
+                               sizeof(data));
+  WriteAndWaitForACK(message, env);
+
+  // The ACK byte was already read, read remaining bytes of header
+  EosFlightBlockResponseHeader header;
+  port.FullRead(
+    std::span{ reinterpret_cast<std::byte*>(&header) + 1, sizeof(header) - 1 },
+    env,
+    std::chrono::milliseconds(5000));
+
+  // Check that received ID matches the requested one
+  if (header.block_id != ToLE16(block_id))
+    throw("Bad ID");
+
+  block.ResizeDiscard(FromLE16(header.block_size));
+
+  port.FullRead(
+    { block.data(), block.size() }, env, std::chrono::milliseconds(5000));
+
+  std::byte crc_byte[1];
+  port.WaitAndRead(crc_byte, env, std::chrono::milliseconds(5000));
+
+  // Message has 3 parts: header, block, and CRC byte.
+  // Calculate CRC from all 3 parts, result should be 0x00
+  uint8_t crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&header), sizeof(header), 0xFF, 0x69);
+  crc = Calculate8bitCRC(block.data(), block.size(), crc, 0x69);
+  if (Calculate8bitCRC(crc_byte, 1, crc, 0x69) != 0)
+    throw("Bad CRC");
```

Comment:
Don't throw a C string; also the parentheses are unnecessary. `throw` is not a function.